block-producer/src/batch_builder/batch.rs

```diff
@@ -66,52 +64,32 @@ impl TransactionBatch {
         let produced_nullifiers =
             txs.iter().flat_map(|tx| tx.input_notes().iter()).cloned().collect();
 
-        let (created_notes, created_notes_smt, created_note_envelopes) = {
-            let created_note_envelopes: Vec<NoteEnvelope> = txs
+        let (created_note_envelopes_with_details, created_notes_smt) = {
+            let created_note_envelopes_with_details: Vec<(NoteEnvelope, Option<Vec<u8>>)> = txs
```

Comment:
You would leave the outer var and struct name the same but only change the inner var to something shorter right?

Reply:
In the future I would go through the files to refactor, I think we should simplify this concept of what is a Note:
- Note
- NoteEnvelope
- CreatedNote
- ect...

For now this would require a too large refactor to do in this PR. 

I have changed the name simply to `created_note` as an internal var it stays readable.